packages/typegpu/src/name.ts

```diff
@@ -1,9 +1,19 @@
+import { ArgNames, Block } from 'tinyest';
 import { $getNameForward } from './shared/symbols.ts';
 
 interface MetaData {
   name: string | undefined;
 }
 
+export interface FunctionMetaData extends MetaData {
+  ast: {
+    argNames: ArgNames;
+    body: Block;
+    externalNames: string[];
+  };
+  externals?: Record<string, unknown> | undefined;
+}
```

Comment:
I think we can store all metadata on a single object, as optional keys & values. It's going to be easier to add existing metadata to new types of resources, and new types of metadata.

```ts

export interface MetaData {
  name?: string | undefined;
  ast?: {
    argNames: ArgNames;
    body: Block;
    externalNames: string[];
  } | undefined;
  externals?: Record<string, unknown> | undefined;
}
```